src/components/cc-logs-instances/cc-logs-instances.js

```diff
@@ -0,0 +1,755 @@
+          commit: instances[0].deployment.commitId,
+        })}
+        <div class="section-content instances">
+          ${this._renderInstances(instances, false)}
+        </div>
+      </fieldset>
+    `;
+  }
+
+  /**
+   * @param {Array<Instance>} instances
+   */
+  _renderDeletedInstances (instances) {
+    if (instances.length === 0) {
+      return '';
+    }
+
+    return html`
+      <fieldset class="section section--deleted">
+        ${this._renderHeader({
+          title: i18n('cc-logs-instances.deleted.header'),
+          icon: iconHeaderDeleted,
+        })}
+        ${this._renderInstancesGroupedByDeployment(instances, false)}
+      </fieldset>
+    `;
+  }
+
+  /**
+   * @param {Array<GhostInstance>|null} instances
+   */
+  _renderGhostInstances (instances) {
+    if (instances == null || instances.length === 0) {
+      return '';
+    }
+
+    return html`
+      <fieldset class="section section--ghost">
+        ${this._renderHeader({
+          title: i18n('cc-logs-instances.ghost.header'),
+          icon: iconHeaderGhost,
+        })}
+
+        <div class="section-content">
+          <cc-notice intent="info" message="${i18n('cc-logs-instances.ghost.notice')}"></cc-notice>
+          <div class="instances instances--ghost">
+            ${instances.sort(GHOST_INSTANCE_SORT_ORDER).map((instance) => html`
+            <label class="instance" for="${instance.id}">
+              <input type="checkbox"
+                     id="${instance.id}"
+                     .checked=${this._isSelected(instance.id)}
+                     @change=${this._onInstanceClick}>
+              <span class="instance-id">${instance.id}</span>
+            </label>
+          `)}
+          </div>
+        </div>
+      </fieldset>
+    `;
+  }
+
+  _renderHeader ({ title, icon, commit }) {
+    return html`
+      <legend class="section-header">
+        ${icon != null ? html`
+          <cc-icon class="section-header-icon" .icon=${icon}></cc-icon>
+        ` : ''}
+        <span class="section-header-title">${title}</span>
+        ${commit != null ? this._renderCommit(commit) : ''}
+      </legend>
+    `;
+  }
+
+  /**
+   * @param {Array<Instance>} instances
+   * @param {boolean} renderState
+   */
+  _renderInstancesGroupedByDeployment (instances, renderState) {
+    const groups = groupBy(instances, (instance) => instance.deployment?.id);
+
+    return html`
+      <div class="section-content deployments">
+        ${Object.entries(groups)
+          .sort(([, instances1], [, instances2]) => instances2[0].deployment.creationDate - instances1[0].deployment.creationDate)
+          .map(([_, instances]) => {
+            return html`
+            <fieldset class="deployment">
+              ${this._renderDeploymentDetails(instances[0].deployment)}
+              <div class="instances">
+                ${this._renderInstances(instances, renderState)}
+              </div>
+            </fieldset>
+          `;
+          })}
```

Comment:
suggestion: in this kind of template, removing the arrow body simplifies the code (IMO)

```suggestion
          .map(([_, instances]) => html`
            <fieldset class="deployment">
              ${this._renderDeploymentDetails(instances[0].deployment)}
              <div class="instances">
                ${this._renderInstances(instances, renderState)}
              </div>
            </fieldset>
          `)}
```